### PR TITLE
Add Docs for TripNotification

### PR DESCRIPTION
**Pull request by Lebedinsky**
## Description

Added docs for `libtrip-notification` module classes ([issue](https://github.com/mapbox/navigation-sdks/issues/219))

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NavigationNotificationProvider.kt [hunk 1]
```diff
@@ -3,7 +3,21 @@ package com.mapbox.navigation.trip.notification
 import android.app.Notification
 import androidx.core.app.NotificationCompat
 
+/**
+ * Provide notification for navigation process launched with a [TripSession]
```
Comment (evabishchevich) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NavigationNotificationProvider.kt]: provides?

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt [hunk 2]
```diff
@@ -156,13 +175,28 @@ class MapboxTripNotification constructor(
         }
     }
 
+    /**
+     * Creates [PendingIntent] for open application when notification view is clicked
```
Comment (evabishchevich) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]: I suggest changing to
_Creates [PendingIntent] to open an application when ..._
or
_Creates [PendingIntent] for opening an application when ..._

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NavigationNotificationProvider.kt [hunk 3]
```diff
@@ -3,7 +3,21 @@ package com.mapbox.navigation.trip.notification
...
+ *
+ * @since 1.0.0
+ */
 object NavigationNotificationProvider {
+
+    /**
+     * Build [Notification] based on [NotificationCompat.Builder] params
```
Comment (evabishchevich) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NavigationNotificationProvider.kt]: Let's have _Builds_ to be consistent in this PR

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt [hunk 4]
```diff
@@ -156,13 +175,28 @@ class MapboxTripNotification constructor(
...
+     *
+     * @param applicationContext is [Context]
+     * @return [PendingIntent] for open application
```
Comment (evabishchevich) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]: to open / for opening

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt [hunk 5]
```diff
@@ -156,13 +175,28 @@ class MapboxTripNotification constructor(
...
+     */
     private fun createPendingOpenIntent(applicationContext: Context): PendingIntent? {
         val pm = applicationContext.packageManager
         val intent = pm.getLaunchIntentForPackage(applicationContext.packageName) ?: return null
         intent.setPackage(null)
         return PendingIntent.getActivity(applicationContext, 0, intent, 0)
     }
 
+    /**
+     * Creates [PendingIntent] for finish navigation process ([TripSession]) when
```
Comment (evabishchevich) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]: same as above

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripNotification.kt [hunk 6]
```diff
@@ -3,6 +3,12 @@ package com.mapbox.navigation.base.trip
 import android.app.Notification
 import com.mapbox.navigation.base.trip.model.RouteProgress
 
+/**
+ * Defines notification showing to user during navigation process
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripNotification.kt]:
```suggestion
 * Defines a contract for [Notification] instance provider and manager. This notification is going to be used in a foreground service managed by a [TripSession]
```

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripNotification.kt [hunk 7]
```diff
@@ -3,6 +3,12 @@ package com.mapbox.navigation.base.trip
...
+ * launched with a [TripSession].
+ *
+ * @since 1.0.0
```
Comment (LukasPaczos) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripNotification.kt]: Same as everywhere else - possibly we can skip these.

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt [hunk 8]
```diff
@@ -136,6 +150,11 @@ class MapboxTripNotification constructor(
         return builder
     }
 
+    /**
+     * Creates notification view for collapsed and expanded states
```
Comment (LukasPaczos) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]:
```suggestion
     * Creates live and interactive views displayed in the notification's layout.
```

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt (hunk 5 above)
Comment (LukasPaczos) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]: This doesn't finish the navigation process per se. We might as well be in a free drive mode, without active guidance and this basically stops the trip session.

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NavigationNotificationProvider.kt [hunk 9]
```diff
@@ -3,7 +3,21 @@ package com.mapbox.navigation.trip.notification
 import android.app.Notification
 import androidx.core.app.NotificationCompat
 
+/**
+ * Provide notification for navigation process launched with a [TripSession]
+ *
+ * @since 1.0.0
+ */
 object NavigationNotificationProvider {
```
Comment (LukasPaczos) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NavigationNotificationProvider.kt]: This class should be internal and doesn't require public docs. We still can include guidelines and docs for ourselves though :+1:

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NotificationAction.kt [hunk 10]
```diff
@@ -1,5 +1,12 @@
 package com.mapbox.navigation.trip.notification
 
+import com.mapbox.navigation.base.trip.TripNotification
+
+/**
+ * Defines set of actions available for [TripNotification]
+ *
+ * @since 1.0.0
+ */
 enum class NotificationAction {
```
Comment (LukasPaczos) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NotificationAction.kt]: I believe this class should be internal as well.

Comment (LukasPaczos) [reply libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/NotificationAction.kt]: Nvm, it should be public.

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripNotification.kt (hunk 7 above)
Comment (Lebedinsky) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/TripNotification.kt]: We can, but I think it is good to mark the version of SDK when this or some other classes/methods comes

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt [hunk 11]
```diff
@@ -156,13 +171,26 @@ class MapboxTripNotification constructor(
...
     private fun createPendingOpenIntent(applicationContext: Context): PendingIntent? {
         val pm = applicationContext.packageManager
         val intent = pm.getLaunchIntentForPackage(applicationContext.packageName) ?: return null
         intent.setPackage(null)
         return PendingIntent.getActivity(applicationContext, 0, intent, 0)
     }
 
+    /**
+     * Creates [PendingIntent] for stopping [TripSession] when
+     * proper button is clicked in the navigation view
```
Comment (LukasPaczos) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/MapboxTripNotification.kt]:
```suggestion
     * proper button is clicked in the notification view
```

**Pull request merged**